pkg/apis/workload/v1alpha1/types.go

```diff
@@ -161,4 +161,15 @@ const (
 	// synced from the APIExport to all the APIBindings bound to this APIExport. The workload scheduler will
 	// check all the APIBindings with this annotation for scheduling purpose.
 	ComputeAPIExportAnnotationKey = "extra.apis.kcp.io/compute.workload.kcp.io"
+
+	// UpsyncDerivedResourcesAnnotationKey is an annotation set on a SyncTarget resource that contains a
+	// command-separated list of stringified GroupResource (<resource>.<group>) for the derived resources
+	// that are expected to be upsynced.
+	// To allow upsyncing the endpoints resource related to a synced service, the service should be annotated with:
+	//
+	//   workload.kcp.io/upsync-derived-resources: endpoints
+	//
+	// For now, only endpoints can be upsynced on demand by the syncer with this mechanism,
+	// but the list of such resources will increase in the future.
+	UpsyncDerivedResourcesAnnotationKey = "workload.kcp.io/upsync-derived-resources"
```

Comment:
is this experimental and going to move into the SyncTarget spec?

Reply:
No it's not experimental.
And it's not expected to be added to the SyncTarget (obviously it is a copy/paste erorr in the comment).
The real comment should be:
```
	// UpsyncDerivedResourcesAnnotationKey is an annotation set on synced resources that contains a
	// command-separated list of stringified GroupResource (<resource>.<group>) for the derived resources
	// that are expected to be upsynced.
	// To allow upsyncing an Endpoints resource related to a synced service, the Service instance should be annotated with:
	//
	//   workload.kcp.io/upsync-derived-resources: endpoints
	//
	// For now, only endpoints can be upsynced on demand by the syncer with this mechanism,
	// but the list of such resources will increase in the future
```

Reply:
I fixed the comment in commit 5ba3c356469a5a968009cdd20ac69fc0f7620251

Reply:
> and is this annotation user-facing in the first place? Who should be able to change it?
Yes, it is user-facing. For now the use should be able to change it. Mid-term it could be done by a dedicated coordination controller.

>(now it is protected through admission).
Are *all* the `workload.kcp.io/xxx` annotation protected ?

Reply:
> Are all the workload.kcp.io/xxx annotation protected ?

yes. You have to opt-out in the admission plugin for that.

> Yes, it is user-facing. For now the use should be able to change it. Mid-term it could be done by a dedicated coordination controller.

So it is experimental?

Reply:
The presence of this annotation isn't sufficient to trigger the Upsync, but it will be used as an additional condition by the logic that exists downstream and that marks resources for Upsync (by setting the `RequestState` label to `Upsync`).
In the case of Endpoints, the controller in the Syncer implements the logic that identifies which Endpoints should be Upsynced, and part of this logic is to look at this annotation on the related Service.
 
In any case, there is also a protection on the Upsyncer virtual workspace side + Syncer about the resource types that can be upsynced: for now the list is fixed: only PVs, Pods and Endpoints can be upsynced.

Reply:
Finally made the annotation experimental and added more comments on the annotation definition.